marlowe-playground-client/src/Marlowe/Parser.purs

```diff
@@ -429,7 +435,7 @@ recContract =
 
   observation' = parseTerm $ atomObservation <|> fix \p -> parens observation
 
-  value' = parseTerm $ atomValue <|> fix (\p -> parens value)
+  value' = parseTerm $ atomValue <|> fix (\p -> parens (value unit))
```

Comment:
Why the unit parameter in value?

Reply:
@palas because Purescript requires strict evaluation, and observation parsers is mutually recursive with value parser, so I needed to force lazy evaluation on it

Reply:
I don't quite understand this either, there are other values that depend on observations and visa-versa, why is `Cond` different?

Reply:
@shmish111 @palas none of values actually depend on observations. Basically I had to do the same trick for Isabelle, Plutus and so on.

Reply:
I don't think I would understand this unless I wrote the code myself but I accept 😄 Can you add a nice comment explaining it as best you can

Reply:
@shmish111 I can add link to stackoverflow answer on this :)

Reply:
yeah, I think that's a good idea

Reply:
@shmish111 Basically I'll add this one to the comments: https://stackoverflow.com/questions/36984245/undefined-value-reference-not-allowed-workaround/36991223#36991223